### ISSUE TITLE
Request - add Keyboard shortcut to show file browser

### ISSUE THREAD

**Issue by plynch-chwy** [jupyterlab/jupyterlab#7547]
It seems once I open the command palette, there isn't a shortcut to view the file explorer again.

**Comment by jasongrout** [comment]
Try `Cmd Shift F` on macOS, `Ctrl Shift F` on windows/linux.

**Comment by plynch-chwy** [comment]
Ahh yes thats it, wasn't searching for the right things in shortcuts. Thanks!

**Comment by jasongrout** [comment]
Where were you searching, just curious? This should have been surfaced to you somehow in some obvious way. I actually went to the source code to find it, which I wouldn't expect a user to typically have to do.

**Comment by jasongrout** [comment]
For example, I thought these @blink1073 made these shortcuts available in the tooltips visible by hovering over the icon in the sidebar, but apparently that doesn't show up anymore. Another place they could be visible is if we added items in the View menu for each sidebar tab - then the shortcut would be available in the View menu, similar to how macOS Preview works:

<img width="275" alt="Screen Shot 2019-11-21 at 7 17 22 AM" src="https://user-images.githubusercontent.com/192614/69350774-fd8b1180-0c2e-11ea-8dd9-3d9fa8f8ed1e.png">

**Comment by jasongrout** [comment]
I just tried searching, and I don't find anywhere in the UX where `Accel Shift F` is noted as the shortcut for showing the file browser. I'm re-opening this issue to figure out where we should expose that shortcut in the UX.

**Comment by jasongrout** [comment]
CC @tgeorgeux - any thoughts?

When @ian-r-rose was working on the View menu, we discussed adding a list of options like is available in the macOS preview menu to show specific side-panel items (see above). I forget why we didn't end up doing that.

**Comment by plynch-chwy** [comment]
Tried a few things before making this issue:

- A long hover over all the icons on the left (the "Commands"/command palette is the only that that showed the short cut)
- Looking in the tool bar options
- Searching command palette for a few things (e.g. "file", "browse")
- Searching the shortcuts editor, though there are lots of shortcuts you can find with "file", so I found `apputils:activate-command-palette` and tried to find a relevant `apputils` or `activate-` command to make use of, but couldn't

**Comment by jasongrout** [comment]
Here is where such things are in VSCode's view menu:

<img width="225" alt="Screen Shot 2019-11-21 at 7 22 42 AM" src="https://user-images.githubusercontent.com/192614/69351281-ea2c7600-0c2f-11ea-8c27-05258e62ccc6.png">

**Comment by jasongrout** [comment]
> * (the "Commands"/command palette is the only that that showed the short cut)

Huh, none of the sidebar icons' tooltips show the short cut for me in jlab 1.2.3, including the command palette.

**Comment by jasongrout** [comment]
> Looking in the tool bar options

What does this mean? Where did you go in the UX to do this?

**Comment by jasongrout** [comment]
> Tried a few things before making this issue:

Just curious, is this the order you tried things in?

**Comment by plynch-chwy** [comment]
> > Looking in the tool bar options
> 
> What does this mean? Where did you go in the UX to do this?

Probably have my terminology wrong here, the "File, Edit, View, ..." bar. Menu bar?

**Comment by jasongrout** [comment]
> Probably have my terminology wrong here, the "File, Edit, View, ..." bar. Menu bar?

Ah, yes, "Menu bar"

**Comment by plynch-chwy** [comment]
> > Tried a few things before making this issue:
> 
> Just curious, is this the order you tried things in?

Roughly, yes.

**Comment by vidartf** [comment]
> Huh, none of the sidebar icons' tooltips show the short cut for me in jlab 1.2.3, including the command palette.

I think there is an existing issue that SVG icons don't show tooltip. I'm using the phone rn, so cannot find it.

**Comment by jasongrout** [comment]
> I think there is an existing issue that SVG icons don't show tooltip. I'm using the phone rn, so cannot find it.

We fixed that a few weeks ago. The problem here is not that the tooltip is not showing (it is). It's just that the tooltip doesn't have the keyboard shortcut.

**Comment by jasongrout** [comment]
This is the issue I was thinking about, where @blink1073 implemented showing a shortcut in the tooltip: https://github.com/jupyterlab/jupyterlab/pull/5268

I'm not sure why that isn't working anymore. Perhaps that issue will give an indication about how to do it again or more broadly for the other tabs.

**Comment by jasongrout** [comment]
There is a *lot* in #5268 that is not relevant to the actual core change. The actual relevant change is https://github.com/jupyterlab/jupyterlab/pull/5268/commits/5be19e890ef93c3de28cd9dc82338317d117239f#diff-e4503750b1a94615aa78476d28331dc4R80-R97

**Comment by jasongrout** [comment]
> Huh, none of the sidebar icons' tooltips show the short cut for me in jlab 1.2.3, including the command palette.

I just realized this is correct for me. I disabled the command palette keyboard shortcut in my settings. When I re-enabled it, it showed up in the tooltip.

So I think the good first issue here is going through the rest of the side panels and doing the same sort of thing to show the keystroke in the tooltip.

**Comment by tgeorgeux** [comment]
In an effort to understand how I could adjust these in the future. So far I have found these things to be true:
- This commit is the code we need to show shortcuts in the tooltip: https://github.com/jupyterlab/jupyterlab/pull/5268/commits/5be19e890ef93c3de28cd9dc82338317d117239f#diff-e4503750b1a94615aa78476d28331dc4R80-R97
- File Browser has a keyboard shortcut which is `Accel Shift F` and the command ID is `filebrowser:toggle-main` this lives in `packages/filebrowser-extension`
- Command Palette has a keyboard shortcut which is `Accel Shift C` with a command ID of `apputils:activate-command-palette` this lives in `packages/apputils-extension`
- I was unable to find a shortcut in existence for 'running', 'cell tools', 'tabs', or 'extension manager' (there was a JSON file in the schema folder for extension manager but no shortcut associated with it).

An interesting thing I noted was that the keyboard shortcuts I assume should be associated with running (Accel Shift R) and tabs (Accel Shift T) are both taken by Chrome. 

Some concerns:
- The inconsistency between `activate-command-palette` and `toggle-main` made it difficult to discover these shortcuts, and not very confident that the others don't exist.
- Should all the left sidebar keyboard shortcuts live in the same place (ie: apputils) or in the same place in their respective extensions? It seems like having that inconsistent doesn't serve the goal of easy extendability.

**Comment by tgeorgeux** [comment]
If we need to generate new keyboard shortcuts is that considered complex enough that the "Good First Issue" tag should be removed?

**Comment by jasongrout** [comment]
How about this issue be constrained to just adding the keyboard shortcut for the filebrowser extension, tied to the `filebrowser:toggle-main` command. Then I think it is a good first issue, at least that part.



> The inconsistency between `activate-command-palette` and `toggle-main` made it difficult to discover these shortcuts, and not very confident that the others don't exist.

Extensions will likely have different command names (though I agree it would be great if it were more consistent with core extensions). In this case, since there can be more than one filebrowser, the filebrowser command has "main" in the title to indicate it is toggling the primary filebrowser. We should check if the command does indeed "activate" or "toggle".



> Should all the left sidebar keyboard shortcuts live in the same place (ie: apputils) 

No. Arbitrary extensions can add sidebar tabs, so they need a way to give a command to activate their widget.

> or in the same place in their respective extensions? 

I'm not sure what you mean by this. An extensions default keyboard shortcuts live in their schema.json settings file, which is the same place, right?

**Comment by tgeorgeux** [comment]
> I'm not sure what you mean by this. An extensions default keyboard shortcuts live in their schema.json settings file, which is the same place, right?

Yes, this is in the same place, but I don't see why just command palette would live in apputils, "if they belong in the extension put them in the extension". I suppose the command palette lives there because there is no 'command palette' extension.

**Comment by jasongrout** [comment]
> Yes, this is in the same place, but I don't see why just command palette would live in apputils

Ah, yes, partly a historical artifact. The thought was that the command palette is pulling in something from the underlying libraries and is integral to the application. It probably could go into its own extension.

**Comment by markellekelly** [comment]
The file browser tooltip should be fixed as of #7781. Should another issue be opened for commands to open the remaining side panels?

**Comment by tgeorgeux** [comment]
@markellekelly Yes please! a command to show/hide sidebars would be great. I would suggest the following:
Show/Hide left sidebar
Show/Hide right sidebar
Show/Hide both sidebars: (Default to showing both if toggled with one open one close, then close both only after both are opened.)

**Comment by jasongrout** [comment]
> Show/Hide left sidebar

Do you mean "Show Left Sidebar", with a toggled state that indicates if it is already showing? I think that's the pattern we've been using, rather than changing the title to "Hide..."

We don't have a good toggle mark for partly true. That would be the hard thing for both sidebars. Or perhaps "Show Both Sidebars" should just be true if both are actually shown? That way it always toggles the value, i.e., if one is open, the command is considered toggled false, so we show both so it appears as true.

**Comment by tgeorgeux** [comment]
I think that sounds great @jasongrout I was addressing the functionality only. 

From a user's perspective, the same hotkey combination should toggle whether the left is shown, the same for the right side and for both sides. I agree that one open one closed should throw a 'false' on showing both sidebars.

**Comment by markellekelly** [comment]
Since the file browser and command palette both have keyboard shortcuts to open their own panel, should other panels (e.g. cell tools, running) have their own commands as well?
